src/wp-includes/kses.php

```diff
@@ -931,19 +932,22 @@ function wp_kses_version() {
  * @global array[]|string $pass_allowed_html      An array of allowed HTML elements and attributes,
  *                                                or a context name such as 'post'.
  * @global string[]       $pass_allowed_protocols Array of allowed URL protocols.
+ * @global bool           $pass_allowed_comments  Whether HTML comments are allowed or not.
  *
  * @param string         $string            Content to filter.
  * @param array[]|string $allowed_html      An array of allowed HTML elements and attributes,
  *                                          or a context name such as 'post'. See wp_kses_allowed_html()
  *                                          for the list of accepted context names.
  * @param string[]       $allowed_protocols Array of allowed URL protocols.
+ * @param bool           $allowed_comments  Whether HTML comments are allowed or not.
  * @return string Content with fixed HTML tags
  */
-function wp_kses_split( $string, $allowed_html, $allowed_protocols ) {
-	global $pass_allowed_html, $pass_allowed_protocols;
+function wp_kses_split( $string, $allowed_html, $allowed_protocols, $allowed_comments ) {
```

Comment:
Because this is a publicly available function, the new argument needs to have a default value for not to cause a crush where it is used without this new attribute, and it needs to keep the behavior that was presented before the change. So, I think it needs: `$allowed_comments = true`

Reply:
it could be helpful to avoid adding another boolean in there as well. with the other allowed values there's perhaps a clearer inherent legibility to their intent, but seeing `true` or `false` at the end of a call can be confusing.

another option here could be to use a basic string value as an enum to self-document.

```php
 * @param string $comments 'preserve' or 'remove' indicating what to do with HTML comment contents.
 */
function wp_kses_split( $string, $allowed_html, $allowed_protocols, $comments = 'preserve' ) {
```